### ISSUE TITLE
[Feature Request]: Lidarr support in nzbToMedia...

### ISSUE THREAD

**Issue by gibxxi** [clinton-hall/nzbToMedia#1350]
QNAP_Stephane has now released a QPKG for Lidarr on the QNAP NAS devices, and the app itself has been in development for some time. While not mature, I'm of the understanding it's stable enough for testing purposes.

So now, the only element missing is integration within nzbToMedia. Sonarr is there, Radarr is there, could we have added support for Lidarr as well please Clinton?

Regards,

Dan / Gib.

**Comment by clinton-hall** [comment]
ok.. I've built most of this, but Looking at the Lidarr API I can't see an option to call a command for the renamer...
So I think this is still something that needs to be added in Lidarr for nzbToMedia to be able to queue the renaming... but it may be that Lidarr don't intend for this to be the way?

**Comment by gibxxi** [comment]
Hi Clinton,

Thanks for taking a look at this. I know that Lidarr has a built-in renamer. Given that music file naming and tagging is a very precise art and that everyone who maintains music libraries has their own "preferences", it wouldn't surprise me. It looks to be that like Sonarr and Radarr, Lidarr does the renaming on import, so adding support for it *may* not be nessecary. It's early days yet, and maybe your assumption is right. Maybe they simply haven't added it to the API yet.

Having said that, since both Radarr and Lidarr are based off Sonarr, I would of thought that this functionality would of already been in the base code, but would just need adjusting for music. Maybe not.

As long as you are aware of it, it's all good. We've waiting this long for it to bevome available, a bit longer won't hurt.

Gib.

**Comment by clinton-hall** [comment]
Basically, I have copied the Radarr/Sonarr code and put in an api call to call the command `DownloadedAlbumScan` (based on the Sonarr `DownloadedEpisodeScan`
however I can't see any documentation to this.
Likewise, I check for CDH at `config/downloadClient` looking for `enableCompletedDownloadHandling` but I see no documentation to this...

In short, the `flow` is added, along with the configurations. But I'll need to see some documentation/guidance to make this work. 

BTW. The Lidarr documentation is good... so I suspect that they actually haven't added this feature, as opposed to having not documented it.

**Comment by gibxxi** [comment]
Hi Clinton, just saw that's you've implemented the config. I've resurrected my "fork" from back a while when we had SSL issues with NZB Get, and merged all updates since then by you into the fork. I've adjusted two typos in the config, rather than submit an issue for something so petty. I've generated a pull request for the changes.

Thanks again for supporting this. I'll update my local Lidarr / nzbToMedia config to do some testing once I have something I can test nzbToMedia (Lidarr) with.
